hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -97,10 +94,10 @@ func (use *UnknownSchemaError) Error() string {
 }
 
 // NewSchemaScanner constructs a new scanner, ready for use
-func NewSchemaScanner(idFactory astmodel.IdentifierFactory, configuration *config.Configuration) *SchemaScanner {
+func NewSchemaScanner(idFactory astmodel.IdentifierFactory, configuration *config.Configuration, typeHandlers map[SchemaType]TypeHandler) *SchemaScanner {
```

Comment:
Why inject the type handlers from outside the type? It feels as though an internal (implementation) issue is being pushed outside, violating encapsulation.

Reply:
1. Originally in my PR you haven't yet seen, I needed to supply different handlers for the "ownership discovery" bits vs the normal bits. Now that this is happening in a pipeline stage this is no longer needed, so technically speaking I can revert this change (it's not required for anything that is coming).
2. I don't think that this necessarily violates encapsulation though - `TypeHandlers` as a property on `SchemaScanner` is already public, as is the function `DefaultTypeHandlers` - even the name `DefaultTypeHandlers` suggests that a consumer of the schema scanner understands that there are `TypeHandlers` and they can be overridden - this is just providing an easier way to do it all on one line as opposed to constructing a `SchemaScanner` and then going `scanner.TypeHandlers[refHandler] = ...`.

With that said -- since this change isn't actually required for what is coming, I've made `DefaultTypeHandlers` package scope rather than exported and removed this parameter since there's no caller who needs to make changes to the type handlers collection right now.